### PR TITLE
Use krun bash script directly for kevm interpret

### PR DESCRIPTION
**Pull request by ehildenb**
This uses the `krun` bash script directly instead of the low-level handling we were doing to invoke `interpreter` instead.

- Remove `kevm interpret --backend java ...` support (can still use `kevm run ...` there).
- Correct the way that `--unparse` and `--no-unparse` are handled (did not work before, now will default to `--unparse` for anything except `kevm interpret` command, and to `--no-unparse` for `kevm interpret` command).
- Adjust `kevm interpret` to call `krun` bash runner script directly, instead of bypassing it an invoking `interpreter`.
- Refactor resulting (now simpler) code for `kevm interpret ...`.

Performance measurements:

- Fastest tests on LLVM backend (before): 30ms
- Introduced time on LLVM backend: +50ms
- LLVM backend Test-suite overall slowdown (`make test-conformance -j20 TEST_CONCRETE_BACKEND=llvm`): 1.44X
- Fastest tests on Haskell backend (before): 15s
- Reduced time on Haskell backend: -7s
- Haskell backend test-suite overall speedup (`make test-vm -j12 TEST_CONCRETE_BACKEND=haskell`): 0.48X

### CODE REVIEW

**Comment by RaoulSchaffranek** [comment]
`make test-vm TEST_CONCRETE_BACKEND=haskell --dry-run` produces the following command which throws an error on my machine.

~~~bash
kevm interpret tests/ethereum-tests/LegacyTests/Constantinople/VMTests/vmBitwiseLogicOperation/sgt2.json  --backend haskell --mode VMTESTS --schedule DEFAULT --chainid 1 
~~~

~~~
Created bug report: kore-exec.tar.gz
[Error] krun: Backend crashed during rewriting with exit code 1
[Error] Critical: Parse error
 (ParseError: ERROR: Unexpected end of file while parsing)
~~~

Adding `-haskell-backend-arg --smt=none` seems to work:

~~~bash
kevm interpret tests/ethereum-tests/LegacyTests/Constantinople/VMTests/vmBitwiseLogicOperation/sgt2.json  --backend haskell --mode VMTESTS --schedule DEFAULT --chainid 1 --haskell-backend-arg --smt=none
~~~

**Review by radumereuta — APPROVED** [review]
The code lgtm.
I tried to test it, but I ran into `./kevm: line 20: kore-json.py: command not found`
I guess I need to install it, but that would require a lot more time.
Let me know if you want me to go through all the steps to make sure everything works.

I would like it if you would get one more approval before merging though.

**Comment by ehildenb** [comment]
I'm going to fix @RaoulSchaffranek 's issue first, and yes, I want it working on your machine first. It should _always_ work on your machine (all the downstream semantics should).

The issue you're having is you're executing the `./kevm` script in the root of the repo, and you should be executing the installed one at `.build/usr/bin/kevm` instead.

**Comment by ehildenb** [comment]
@RaoulSchaffranek your issue with the haskell backend should be fixed now, at least it should error with this warning now:

```
<kevm>                                                              
  <k>      
    project:EthereumSimulation ( { "sgt2" : { "_info" : { "comment" : "" , "filledwith" : "testeth 1.8.0-6+commit.c7ba3b69" , "lllcversion" : "Version: 0.5.14-develop.2019.11.27+commit.8f259595.Linux.g++" , "source" : "src/VMTestsFiller/vmBitwiseLogicOperation/sgt2Filler.js
on" , "sourceHash" : "bb81e3b79fbad22504061aa4955a2dd54b0a04669c75f126fd473015fd340d08" , .JSONs } , "callcreates" : [ .JSONs ] , "env" : { "currentCoinbase" : "0x2adc25665018aa1fe0e6bc666dac8fc2697ff9ba" , "currentDifficulty" : "0x020000" , "currentGasLimit" : "0x7ffffffff
fffffff" , "currentNumber" : "0x01" , "currentTimestamp" : "0x03e8" , .JSONs } , "exec" : { "address" : "0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6" , "caller" : "0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b" , "code" : "0x60007ffffffffffffffffffffffffffffffffffffffffffffffff
fffffffffffffffff1360005500" , "data" : "0x" , "gas" : "0x0186a0" , "gasPrice" : "0x0c" , "origin" : "0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b" , "value" : "0x0b" , .JSONs } , "gas" : "0x01730c" , "logs" : "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d4934
7" , "out" : "0x" , "post" : { "0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6" : { "balance" : "0xd3c21bcecceda1000000" , "code" : "0x60007fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff1360005500" , "nonce" : "0x00" , "storage" : { .JSONs } , .JSONs } , .JSO
Ns } , "pre" : { "0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6" : { "balance" : "0xd3c21bcecceda1000000" , "code" : "0x60007fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff1360005500" , "nonce" : "0x00" , "storage" : { .JSONs } , .JSONs } , .JSONs } , .JSONs 
} , .JSONs } ~> . ) ~> .
  </k>             
  <exit-code>  
    1               
  </exit-code>     
  <mode>     
    VMTESTS         
  </mode>         
  <schedule>    
    DEFAULT           
  </schedule>       
  <ethereum>           
    <evm>            
      <output>   
        b""   
      </output>   
      <statusCode>         
...
```

To get it to make further progress on the Haskell backend (not needed for LLVM backend), you need to apply this patch: https://github.com/runtimeverification/evm-semantics/issues/1291

**Review by RaoulSchaffranek — APPROVED** [review]
Looks good to me.
With the patch from #1291 the conformance tests are passing on both backends.  If I don't apply the patch I get an error message from the Haskell backend but not from the llvm backend.

**Comment by ehildenb** [comment]
Sounds good! Yes, that's how it's supposed to be. We don't include the `driver.md` on Haskell backend by default, because we don't care to include it in proofs (it's just for running general-state-tests conformance test-suite).

**Comment by radumereuta** [comment]
@ehildenb those instructions worked.
the evm-semantics repo could use a bit of a restructuring.
Having separate folders for /src and /scripts could make things look a lot more organized.

**Pull request merged**
